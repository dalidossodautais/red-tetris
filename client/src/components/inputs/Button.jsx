import React from "react";

import Button from "@material-ui/core/Button";
import { makeStyles } from "@material-ui/core/styles";

function useContainedClasses() {
  return makeStyles((theme) => ({
    button: {
      alignSelf: "flex-end",
      backgroundColor: theme.primaryBackgroundColor,
      color: theme.primaryTextColor,
      "&:hover": {
        backgroundColor: theme.primaryBackgroundColor,
        color: theme.primaryTextColor,
        opacity: 0.9,
      },
    },
  }))();
}

function ContainedButton({ children, disabled, onClick }) {
  const classes = useContainedClasses();
  return (
    <Button
      className={classes.button}
      disabled={disabled}
      onClick={onClick}
      variant="contained"
    >
      {children}
    </Button>
  );
}

function useOutlinedClasses() {
  return makeStyles((theme) => ({
    button: {
      borderColor: theme.primaryBackgroundColor,
      borderWidth: "2px",
      color: theme.primaryBackgroundColor,
      width: "49.8%",
    },
  }))();
}

function OutlinedButton({ children, disabled, onClick }) {
  const classes = useOutlinedClasses();
  return (
    <Button
      className={classes.button}
      disabled={disabled}
      onClick={onClick}
      variant="outlined"
    >
      {children}
    </Button>
  );
}

export default function ({ children, disabled, onClick, variant }) {
  if (variant === "contained")
    return (
      <ContainedButton disabled={disabled} onClick={onClick}>
        {children}
      </ContainedButton>
    );
  if (variant === "outlined")
    return (
      <OutlinedButton disabled={disabled} onClick={onClick}>
        {children}
      </OutlinedButton>
    );
  return null;
}
