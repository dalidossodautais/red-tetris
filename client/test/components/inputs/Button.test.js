import React from "react";
import renderer from "react-test-renderer";
import Button from "../../../src/components/inputs/Button";

describe("Button", () => {
  test("Standard", () => {
    const component = renderer.create(<Button />);
    // rendering
    let tree = component.toJSON();
    expect(tree).toBe(null);
    // re-rendering
    tree = component.toJSON();
    expect(tree).toBe(null);
  });
  test("Contained", () => {
    const component = renderer.create(<Button variant="contained" />);
    // rendering
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("Outlined", () => {
    const component = renderer.create(<Button variant="outlined" />);
    // rendering
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
