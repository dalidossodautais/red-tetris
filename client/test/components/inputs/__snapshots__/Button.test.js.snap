// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Contained 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-1"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  />
</button>
`;

exports[`Button Contained 2`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-1"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  />
</button>
`;

exports[`Button Outlined 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-button-2"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  />
  <span
    className="MuiTouchRipple-root"
  />
</button>
`;

exports[`Button Outlined 2`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-button-2"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  />
  <span
    className="MuiTouchRipple-root"
  />
</button>
`;
